pkg/utils/logger/logger.go

```diff
@@ -13,6 +13,9 @@
 
 package logger
 
+// "aws-k8s-agent" string passed as an input by cmd/aws-k8s-agent/main.go.
+// This is then compared to load binary specific configuration in New() method
+// Do not make any changes to "aws-k8s-agent" string.
```

Comment:
This relies on humans reading this comment and maintaining consistency across the two files, so is a bit fragile :(

Can we make the compiler check this for us?  Maybe:

1. In this file, export the constant:  `const IpamdBinaryName = "aws-k8s-agent"`
2. In `cmd/aws-k8s-agent/main.go` do `const binaryName = logger.IpamdBinaryName`

I think this still deserves a comment like yours above on the const in aws-k8s-agent/main.go to indicate that this specific binaryname value is magical in `logger.New()`.

(If we _really_ wanted to clean this up, we should just change `New()` to take a more useful argument directly instead of implicitly via BinaryName, and then pass the ipamd-special logging config directly from ipamd calling code.)

Reply:
Thanks for the feedback, @anguslees. I went ahead made modification to remove Binary name and pass logger config during binary initialization.